src/diem/offchain/types/fund_pull_pre_approval_types.py

```diff
@@ -0,0 +1,68 @@
+import typing
+
+from dataclasses import dataclass, field as datafield
+
+from .command_types import CommandType
+
+
+class FundPullPreApprovalStatus:
+    # Pending user/VASP approval
+    pending = "pending"
+    # Approved by the user/VASP and ready for use
+    valid = "valid"
+    # User/VASP did not approve the pre-approval request
+    rejected = "rejected"
+    # Approval has been closed by the user/VASP and can no longer be used
+    closed = "closed"
+
+
+class TimeUnit:
+    day = "day"
+    week = "week"
+    month = "month"
+    year = "year"
+
+
+@dataclass(frozen=True)
+class CurrencyObject:
+    amount: int
+    currency: str
+
+
+@dataclass(frozen=True)
+class ScopedCumulativeAmountObject:
+    unit: str = datafield(metadata={"valid-values": [TimeUnit.day, TimeUnit.week, TimeUnit.month, TimeUnit.year]})
+    value: int
+    max_amount: CurrencyObject
+
+
+@dataclass(frozen=True)
+class FundPullPreApprovalScopeObject:
+    expiration_timestamp: int
+    max_cumulative_amount: typing.Optional[ScopedCumulativeAmountObject]
+    max_transaction_amount: typing.Optional[CurrencyObject]
```

Comment:
looks like these fields should be inside FundPullPreApprovalObject.
I understand this maybe better for understanding their relationship, but making them different with dip-8 data structure will create confusion and harder to match document in dip-8.

Reply:
It isn't my initiative. There is a pending update for DIP-8 where this is the new structure. I was told the update jas been reviewed and should be the point of reference.

Reply:
It isn't my initiative hehe. There is a pending update for DIP-8 where this is the new structure. I was told the update has been reviewed and should be the point of reference.